### ISSUE TITLE
Negative balance error while running check:data

### ISSUE THREAD

**Issue by Arslan2** [helium/rosetta-helium#49]
I built the docker file for mainnet using `optimizations` branch for rosetta-helium.
While running `check:data` I am facing this issue.

`unable to process blocks: unable to process block: negative balance -300000000:&{Symbol:HNT Decimals:8 Metadata:map[]} for &{Address:1447S3QGHzvXc8PjFwrNsq4FkCG376FDNr8MwAEs5ZgxaSqRZpH SubAccount:<nil> Metadata:map[]} at &{Index:1156261 Hash:aOUCcjukjrg8T5wRiA41CaYERleJlPmQJXZHBn-3cfA}: unable to add block to storage aOUCcjukjrg8T5wRiA41CaYERleJlPmQJXZHBn-3cfA:1156261: unable to sync to 1156560`

**Comment by syuan100** [comment]
We are waiting on some PRs to be merged for `blockchain-node` and `blockchain-core` that should fix this issue. Please keep an eye on the #48 PR.

**Comment by Arslan2** [comment]
@syuan100 Currently running optimization branch and still can see the negative balance issue. Here are the logs.

unable to process blocks: unable to process block: negative balance -123356765:&{Symbol:HNT Decimals:8 Metadata:map[]} for &{Address:141QXEmHUe4NMHDPN9wv97nyV4MQzy9XqLAraqZfVwEzgZztkP9 SubAccount:<nil> Metadata:map[]} at &{Index:1156270 Hash:oVtVaPsbuZkNfq8I1dmLCrd3XGRSnj7RCmiHF25S3g4}: unable to add block to storage oVtVaPsbuZkNfq8I1dmLCrd3XGRSnj7RCmiHF25S3g4:1156270: unable to sync to 1158942

**Comment by syuan100** [comment]
@Arslan2 are you using the correct config? The `start_index` should be 1156322.

1156270 will yield a negative balance because there is not a complete ledger at that block.

**Comment by GeekArthur** [comment]
I will create another issue related to `/block request failed`, for the time being we will keep this open

**Comment by Arslan2** [comment]
@syuan100 on latest optimization branch with `start_index` 1156322, i can still produce the negative balance issue. Here are the logs.

`unable to process blocks: unable to process block: negative balance -28452316:&{Symbol:HNT Decimals:8 Metadata:map[]} for &{Address:<nil> SubAccount:<nil> Metadata:map[]} at &{Index:1157491 Hash:WstLRU9pog1Kat1i6Nh9UC390HcsN5lfbj9zfNjxt_U}: unable to add block to storage WstLRU9pog1Kat1i6Nh9UC390HcsN5lfbj9zfNjxt_U:1157491: unable to sync to 1157725`
